sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1356,4 +1360,36 @@ export class ApplicationService extends RecordDataModelService<Application> {
       }
     }
   }
+
+  /**
+   * Archives one or more applications when 43 days
+   * have passed the end of the study period.
+   * @param auditUserId user making changes to table.
+   */
+  async archiveApplications(auditUserId: number): Promise<void> {
+    const auditUser = { id: auditUserId } as User;
+
+    // Build sql statement to get all application ids to archive
+    const applicationsToArchive = await this.repo
+      .createQueryBuilder("application")
+      .select("application.id")
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .where("application.applicationStatus = :completed")
+      .andWhere(
+        "(CURRENT_DATE - offering.studyEndDate) > :applicationArchiveDays",
+      )
+      .andWhere("application.isArchived <> :isApplicationArchived")
+      .getSql();
+
+    this.repo
+      .createQueryBuilder()
+      .update(Application)
+      .set({ isArchived: true, modifier: auditUser })
+      .where(`applications.id IN (${applicationsToArchive})`)
```

Comment:
may I know the reason why the variable was not set something like this , 
![image](https://user-images.githubusercontent.com/77353155/171918344-6b56917e-b8c0-4e02-91f3-3fd4c8ab72f5.png)

Reply:
Nice catch @ann-aot

Reply:
Yeah I will add.

Reply:
Sorry, I am not following this, the variable is already set as a parameter `.setParameter("completed", ApplicationStatus.completed)`. What is the point here?

Reply:
ignore my comments, i missunderstood

Reply:
@henryoforeh-aot apparently there was a misunderstanding. There is nothing to be changed related to this.

Reply:
Oh I thought your comment was on filtering for applications either in completed or enrollment status.

Reply:
Okay thanks for the clarification.